ksqldb-engine/src/main/java/io/confluent/ksql/internal/PullQueryExecutorMetrics.java

```diff
@@ -202,6 +232,15 @@ public void recordRowsReturned(
     }
   }
 
+  public void recordZeroRowsReturnedForError() {
```

Comment:
I also hard-coded the "zero", since that's what we would always record for errors.

Reply:
Thanks. Just to make sure we're on the same page, I'm just preserving and making explicit the existing logic. I did think it would make more sense not to report these metrics in error cases, but I just left it alone.

If you think it's fine (and maybe better) not to report these metrics at all for errors, we can clean things up even more.